src/relax/backend/vm/vm_builtin_lower.cc

```diff
@@ -81,35 +83,59 @@ class VMBuiltinLowerMutator : public ExprMutator {
       }
       return ShapeExpr({ret});
     } else {
-      return Call(builtin_compute_alloc_shape_, {shape, DataTypeImm(dtype)}, Attrs(),
-                  {GetStructInfo(shape)});
+      // TODO(@slyubomirsky): Find a way to register these builtins as pure to avoid needing to emit
+      // call_pure each time
+      return WrapCallPure(Call(builtin_compute_alloc_shape_, {shape, DataTypeImm(dtype)}, Attrs(),
+                               {GetStructInfo(shape)}));
```

Comment:
An example (one of many) of having to insert `call_pure` because of the use of `PackedFunc`s at low levels of compilation.

Reply:
No longer needed with `RemovePurityChecking`